### DISCUSSION TITLE
Load JS code from wheel?

### DISCUSSION THREAD

**Discussion by dnalor123** [pyodide/pyodide#4627] [Q&A]
I have a wheel that is built with Pyodide (in tree). Now I'd like the wheel's Python code to call some JavaScript functions via Pyodide's `js` module.

Is there a way to include JavaScript files in a wheel? How would one load/import these?

For what it's worth, I'm using Pyodide as a Jupyter Lite kernel, so it runs in a web worker.

**Comment by ryanking13** [comment]
You can read the JS file and run it with `pyodide.code.run_js`.

```py
import pathlib
js_code = pathlib.Path("path_to_the_js_file.js").read_text()

import pyodide.code
pyodide.code.run_js(js_code)
```

Otherwise I think you could try something like this: https://stackoverflow.com/questions/57121467/import-a-module-from-string-variable

**Reply by dnalor123** [reply]
Thanks for the answer. Although this works, I actually want to import JavaScript *modules*.

Trying to follow your suggestion:

```Python
abc = await js.eval("(async () => { return await import('abc.js') })()")
```

But now `abc` is a promise. Is there a way set this up so that one can call `abc.f()` instead of `(await abc).f()`?

**Reply by ryanking13** [reply]
As the import method works asynchronous, for now, it is not possible to use it synchronously in Python. With JavaScript Promise Integration, I think you can do it, but it is only available under experimental flag in chrome.
